presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -335,6 +335,18 @@ void PrestoServer::run() {
           proxygen::ResponseHandler* downstream) {
         server->reportServerInfo(downstream);
       });
+  if (systemConfig->enableRuntimeMetricsCollection()) {
```

Comment:
Can we move this to the `registerRuntimeMetricsReporters` implementation?

Reply:
shouldn't it be that when config is enabled we register the reporters and the http endpoint.

Reply:
I feel it is cleaner if the http endpoints are added by the reporter itself.

Reply:
We decided to use a reporter-specific endpoint anyway.
example: `/v1/info/prometheus-metrics/`